v2/sender.go

```diff
@@ -29,6 +30,7 @@ type AzServiceBusSender interface {
 // Sender contains an SBSender used to send the message to the ServiceBus queue and a Marshaller used to marshal any struct into a ServiceBus message
 type Sender struct {
 	sbSender AzServiceBusSender
```

Comment:
document that any usage of sbSender must consider the mutex and take the appropriate lock.